components/shared/DataTable.tsx

```diff
@@ -30,60 +41,73 @@ import {
 import { Column, FilterOption } from "@/types/data-table";
 
 interface DataTableProps<T> {
-  data: T[];
+  data?: T[];
+  apiUrl: string;
   columns: Column<T>[];
   searchFields?: (keyof T)[];
   filterOptions?: FilterOption<T>[];
   itemsPerPage?: number;
+  onDelete: (id: string | number) => Promise<void>;
 }
 
 export function DataTable<T extends { id: string | number }>({
-  data,
+  apiUrl,
   columns,
   searchFields = [],
   filterOptions = [],
-  itemsPerPage = 10,
+  itemsPerPage = 5,
+  onDelete,
 }: DataTableProps<T>) {
+  const [data, setData] = useState<T[]>([]);
+  const [totalItems, setTotalItems] = useState(0);
   const [searchQuery, setSearchQuery] = useState("");
   const [currentPage, setCurrentPage] = useState(1);
   const [filters, setFilters] = useState<string[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [deleteDialog, setDeleteDialog] = useState<{
+    open: boolean;
+    id: string | number | null;
+  }>({ open: false, id: null });
 
-  const filteredData = useMemo(() => {
-    return data.filter((item) => {
-      const matchesSearch = searchFields.some((field) =>
-        String(item[field]).toLowerCase().includes(searchQuery.toLowerCase())
-      );
-      const matchesFilters =
-        filters.length === 0 ||
-        filters.some(
-          (filter) =>
-            String(
-              item[
-                filterOptions.find((option) => option.label === filter)
-                  ?.key as keyof T
-              ]
-            ).toLowerCase() === filter.toLowerCase()
-        );
-      return matchesSearch && matchesFilters;
-    });
-  }, [data, searchQuery, filters, searchFields, filterOptions]);
-
-  const totalPages = Math.ceil(filteredData.length / itemsPerPage);
-  const currentData = filteredData.slice(
-    (currentPage - 1) * itemsPerPage,
-    currentPage * itemsPerPage
-  );
+  const fetchData = async () => {
+    setIsLoading(true);
+    try {
+      const response = await axios.get(apiUrl, {
```

Comment:
💡we don't use axios any more 
💡use base api request route for handling api request